app/forms/candidate_interface/science_gcse_grade_form.rb

```diff
@@ -0,0 +1,218 @@
+module CandidateInterface
+  class ScienceGcseGradeForm
+    include ActiveModel::Model
+    include ValidationUtils
+
+    attr_accessor :grade,
+                  :grades,
+                  :award_year,
+                  :qualification,
+                  :subject,
+                  :other_grade,
+                  :single_award_grade,
+                  :double_award_grade,
+                  :gcse_science,
+                  :biology_grade,
+                  :physics_grade,
+                  :chemistry_grade
+    validates :biology_grade, presence: true, on: :grades, if: :triple_award?
+    validates :chemistry_grade, presence: true, on: :grades, if: :triple_award?
+    validates :physics_grade, presence: true, on: :grades, if: :triple_award?
+    validates :other_grade, presence: true, if: :grade_is_other?
+    validates :grade, length: { maximum: 6 }, unless: :international_gcses_flag_active?, on: :grade
+    validate :grade_length, on: :grade
+    validate :validate_grade_format, unless: :new_record?, on: :grade
+    validate :triple_award_grades_format, if: :triple_award?, on: :grade
+
+    class << self
+      def build_from_qualification(qualification)
+        if FeatureFlag.active?('international_gcses') && qualification.qualification_type == 'non_uk'
+          new(
+            grade: qualification.set_grade,
+            other_grade: qualification.set_other_grade,
```

Comment:
Would be a good idea to move these `set_` methods into this form and give them a different name. They aren't used anywhere else, and neither of them set anything - they just map db values to radio button values as far as I can tell.

Reply:
Yep good idea

Reply:
😞 These `set_` methods are in fact been called by the `GcseQualificationDetailsForm` which is still used in a number of places. I think we should deal with this in a ‘cleanup’ PR once the multi GCSE English and Science flow is shipped